### DISCUSSION TITLE
팀에 컨벤션은 왜 필요할까?

### DISCUSSION THREAD

**Discussion by nayonsoso** [solid-connection/solid-connect-server#135] [논의]
컨벤션을 지키지 않으면 어떤 어려움이 있을까요?

**Comment by Gyuhyeok99** [comment]
컨벤션은 팀의 개발 생산성을 높이고 협업을 원활히 하기 위한 중요한 기준이라고 생각합니다. 이는 단순히 "규칙을 따르는 것" 이상의 의미를 가지며, 팀원 모두가 동일한 의도로 개발할 수 있게 도와줍니다.

### 컨벤션을 지키지 않으면 발생하는 어려움
1️⃣ 코드 이해도 저하
서로 다른 스타일로 작성된 코드가 섞이면 코드의 의도를 파악하기 어렵고, 리뷰 시간과 수정에 걸리는 시간이 늘어남

2️⃣ 협업 효율 저하
컨벤션이 없으면 팀원 간 코드 작성 방식이 달라져 의사소통 비용이 증가하고, 협업 과정에서 불필요한 갈등이 생길 수 있음

3️⃣ 프로젝트 품질 저하
일관성이 없는 코드는 테스트와 디버깅이 복잡해지고, 결과적으로 유지보수에 많은 시간을 소모하게 됩니다. 이 소프트웨어 품질과에 부정적인 영향을 미침

위와 같은 어려움들이 있지만 개인적으로 컨벤션이 제대로 안되어있으면 개발할 때 뭔가 의욕이 조금 떨어지는 거 같습니다 ㅎㅎ..

**Comment by nayonsoso** [comment]
**1/ 개발자들이 코드를 읽고 쓰고 이해하는 시간을 줄여줍니다.**
코드를 작성하는 사람 입장에서, 형식에 대한 의사 결정을 하지 않아도 됩니다.
코드를 리뷰하는 사람 입장에서, 반드시 필요한 곳에만 집중해서 리뷰할 수 있습니다.
특히 코드리뷰는 빠르게 이뤄져야하고, 해당 PR이 목표로 하는 것에만 집중되어야 한다고 생각합니다.
컨벤션이 있으면 전반적으로 생산력과 유지보수성을 좋게 해줍니다.

**2/ 컨벤션을 지키면 좋은 코드가 되는 경우가 많습니다! 👀**
컨벤션은 단순히 '형식'을 정하는 것 이상의 의미를 갖는다고 생각하는데요,
예를 들어 와일드카드(*)를 사용하지 않는다는 컨벤션은 어떤 것들을 import 하는지 분명히 알 수 있게 해줍니다.
하나의 함수가 10줄이 넘어가지 않게 컨벤션을 정한다면, 이를 지키기 위해서 함수를 분리하는 등의 노력을 하게됩니다.
접근제어자를 반드시 명시하자는 컨벤션이 있다면, 접근하면 안되는 패키지를 접근하는 등의 실수를 방지할 수도 있습니다.

**주의해야 할 부분 : 반드시 필요한 부분이 아니라면 넘어가기**
사실 이 부분은 저도 지키기 힘든 부분인데요..!
그렇다고 모든 곳에 컨벤션을 정하고, 이를 지키는지 엄격하게 정할 필욘 없다고 생각합니다.
그게 오히려 리소스가 될 수도 있기 때문..
그렇기 때문에 타협을 할 수 있는 부분이라면 넘어가는 것도 협업에 필요하다고 생각합니다!
